src/json.hpp

```diff
@@ -1476,11 +1707,7 @@ class basic_json
 
     @since version 2.0.0
     */
-    template<typename CompatibleNumberUnsignedType, typename std::enable_if <
-                 std::is_constructible<number_unsigned_t, CompatibleNumberUnsignedType>::value and
-                 std::numeric_limits<CompatibleNumberUnsignedType>::is_integer and
-                 not std::numeric_limits<CompatibleNumberUnsignedType>::is_signed,
-                 CompatibleNumberUnsignedType>::type = 0>
+    template<typename CompatibleNumberUnsignedType, enable_if_t<detail::is_compatible_unsigned_integer_type<number_unsigned_t, CompatibleNumberUnsignedType>::value, int> = 0>
```

Comment:
why do you need to pass `number_unsigned_t` to `detail::is_compatible_unsigned_integer_type` ? Shouldn't passing it `CompatibleNumberUnsignedType` be enough? (as in, can't we simplify this a bit?)